### PR TITLE
Rich/cas 217 fix snapshot balance vote in `createVoteForProposal` controller

### PR DESCRIPTION
**Pull request by rvmelkonian**
This is a subticket of cas-216. Some logic was missing in the Strategy `FetchBalance` method. Also now dummy balance data is returned for local dev.

### CODE REVIEW

**Comment by dbslone** [comment]
We will forget in the future why there is fragmentation in the code. Can you write a quick doc in `docs/<filename>.md` that explains how this operates? Doesn't need to be detailed just need a quick high level explanation for now.

**Comment by germanurrus** [comment]
I tested locally with FE creating and voting a proposal with `token-weighted-default` and works fine

File: backend/main/shared/snapshot-flow.go [hunk 1]
```diff
@@ -23,13 +23,35 @@ type Snapshot struct {
 	Finished     time.Time `json:"finished"`
 }
 
+// copied this quick and dirty from models to get around circular dependency
+// move to shared?
+type Balance struct {
```
Comment (jacksonConrad) [inline backend/main/shared/snapshot-flow.go:28]: Yeah probably cleanest to move this to `shared/structs.go` and use it in `balance.go` and `snapshot-flow.go` if that works

File: backend/main/server/app.go [hunk 2]
```diff
@@ -547,11 +547,13 @@ func (a *App) createVoteForProposal(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// check that proposal is live
-	if !p.IsLive() {
-		err = errors.New("user cannot vote on inactive proposal")
-		log.Error().Err(err)
-		respondWithError(w, http.StatusInternalServerError, err.Error())
-		return
+	if os.Getenv("APP_ENV") != "DEV" {
```
Comment (jacksonConrad) [inline backend/main/server/app.go:550]: is this just to make testing locally easier?

**Review by jacksonConrad — CHANGES_REQUESTED** [review]

**Comment by rvmelkonian** [comment]
clsoing this as it was merged into `cas-172` which has now been merged to main.

**Pull request closed without merging**